### PR TITLE
[ENG-3193] refactor: Refactor away from selectedOrdinal

### PR DESCRIPTION
**Pull request by terencehh**
# 🔘 PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Enhancement
- [x] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

# 📜 Background
There is a state bug where `selectedOrdinal` has improper state when on multiple tabs (it overwriting the current `selectedOrdinal` on the page). This PR introduces a fix by removing state management for `selectedOrdinal`, and instead always opting to look up the ordinal using the `ordinalId` in URL params, and finding the selected ordinal via react-query hooks (with caching)

Issue Link: [ENG-3193](https://linear.app/xverseapp/issue/ENG-3193/refactor-away-from-selectedordinal-and-selectedraresat)
Context Link (if applicable):

# 🔄 Changes
- removed `selectedOrdinal` state actions, reducers, and setters
- updated all `selectedOrdinal` usage to call react-query hook which retrieves ordinal details via the `ordinalId` supplied in URL params

Impact:
- fixes state management bugs when on multiple tabs in send ordinals flow (sendOrdinal & confirmOrdinalTransaction screens)
- indirectly improves SEO since URL contains data now

# 🖼 Screenshot / 📹 Video

Send/Confirm Ordinals Screen Bug video

https://github.com/secretkeylabs/xverse-web-extension/assets/30040178/4699cbe7-4718-4a61-84b0-cdaa313a1412

Send/Confirm Ordinals Screen Fix video

https://github.com/secretkeylabs/xverse-web-extension/assets/30040178/ca6f21dc-aecf-4a55-943a-679429a84938

Recover Funds/Ordinals Screen Bug video

https://github.com/secretkeylabs/xverse-web-extension/assets/30040178/d010e5d2-b53c-43b3-a35a-4594a583e6c9

Recover Funds/Ordinals Screen Fix video

https://github.com/secretkeylabs/xverse-web-extension/assets/30040178/96f1a5ec-4db0-48b2-9b91-77d452f00f4e

# ✅ Review checklist
Please ensure the following are true before merging:

- [x] Code Style is consistent with the project guidelines.
- [x] Code is readable and well-commented.
- [x] No unnecessary or debugging code has been added.
- [x] Security considerations have been taken into account.
- [x] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [x] Code does not introduce new technical debt or issues.

### CODE REVIEW

**Review by jordankzf — DISMISSED** [review]
Well-done! Thank you.

File: src/app/screens/restoreFunds/restoreOrdinals/index.tsx [hunk 1]
```diff
@@ -138,42 +137,37 @@ function RestoreOrdinals() {
...
   return (
     <>
       <TopRow title={t('RESTORE_ORDINAL_SCREEN.TITLE')} onClick={handleOnCancelClick} />
       <Container>
-        {!ordinals ? (
+        {ordinalsQuery.isLoading ? (
           <LoaderContainer>
             <MoonLoader color="white" size={25} />
           </LoaderContainer>
+        ) : ordinalsQuery.ordinals!.length > 0 ? (
```
Comment (teebszet) [inline src/app/screens/restoreFunds/restoreOrdinals/index.tsx:148]: probably don't want to use the ! operator here. use ?. instead to fall through to the false case

File: src/app/screens/restoreFunds/restoreOrdinals/ordinalRow.tsx [hunk 2]
```diff
@@ -62,42 +62,43 @@ interface Props {
...
   isLoading: boolean;
   disableTransfer: boolean;
-  handleOrdinalTransfer: (ordinal: BtcOrdinal, ordinalData: Inscription) => Promise<void>;
+  handleOrdinalTransfer: (ordinal: BtcOrdinal) => Promise<void>;
 }
 
 function OrdinalRow({ ordinal, isLoading, disableTransfer, handleOrdinalTransfer }: Props) {
   const { t } = useTranslation('translation');
-  const ordinalData = useInscriptionDetails(ordinal.id);
+  const inscriptionQuery = useInscriptionDetails(ordinal.id);
```
Comment (teebszet) [inline src/app/screens/restoreFunds/restoreOrdinals/ordinalRow.tsx:70]:
```suggestion
```

File: src/app/screens/restoreFunds/restoreOrdinals/ordinalRow.tsx [hunk 3]
```diff
@@ -62,42 +62,43 @@ interface Props {
...
 
   const onClick = async () => {
-    if (ordinalData && ordinalData.data) {
-      await handleOrdinalTransfer(ordinal, ordinalData.data);
+    if (inscriptionQuery && inscriptionQuery.data) {
+      await handleOrdinalTransfer(ordinal);
```
Comment (teebszet) [inline src/app/screens/restoreFunds/restoreOrdinals/ordinalRow.tsx:74]:
```suggestion
    if (ordinal) {
      await handleOrdinalTransfer(ordinal);
```

Comment (teebszet) [inline src/app/screens/restoreFunds/restoreOrdinals/ordinalRow.tsx:74]: thanks for taking care to keep the existing behaviour here

File: src/app/screens/rareSatsBundle/index.tsx [hunk 4]
```diff
@@ -221,10 +221,6 @@ function RareSatsBundle() {
 
   const isEmpty = !bundle?.satRanges?.length;
 
-  const goBackText = selectedOrdinal?.id
-    ? t('SEND.MOVE_TO_ASSET_DETAIL')
-    : t('NFT_DETAIL_SCREEN.MOVE_TO_ASSET_DETAIL');
-
```
[the remark this answers is not in the document]
Comment (terencehh) [reply src/app/screens/rareSatsBundle/index.tsx:227]: from my testing & understanding, seems like `selectedOrdinal.id` will never be a valid case, so i removed the conditional, correct me if im wrong 🙏

[the remark this answers is not in the document]
Comment (teebszet) [reply src/app/screens/rareSatsBundle/index.tsx:227]: yeah I think selectedOrdinal.id was set if it came from the useOrdinalDetail.ts L111

**Comment by teebszet** [comment]
nice! lgtm 
image

File: src/app/screens/rareSatsBundle/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (terencehh) [reply src/app/screens/rareSatsBundle/index.tsx:227]: after clarification, basically its that we can navigate to `rare-sats-bundle` from ordinal collections - use `location.state.source === 'OrdinalDetail` logic

**Review by teebszet — APPROVED** [review]

**Review by teebszet — DISMISSED** [review]

**Review by jordankzf — APPROVED** [review]

**Review by jordankzf — DISMISSED** [review]

**Comment by DuskaT021** [comment]
@terencehh there is a small conflict that needs to be resolved

**Comment by DuskaT021** [comment]
@terencehh tested, noice, lgtm 👏

**Review by teebszet — APPROVED** [review]

**Pull request merged**